test: snapshot tests for `storage_types` contract dryrun results

peetzweg opened the pull request:
adds snapshot testing of the dryrun results of the functions from the "storage_types" contract.
Contract source can be found here: https://github.com/paritytech/ink/pull/1762

<img width="456" alt="Screenshot 2023-04-27 at 15 47 15" src="https://user-images.githubusercontent.com/839848/234882041-2646b437-bfff-4a14-976c-6da132bd2d1e.png">

peetzweg commented:
The decoded values for `Option<Some>` and `Option<None>` look wrong to me, wdyt @statictype ?

https://github.com/paritytech/contracts-ui/pull/470/files#diff-5558a07e3fdf5db8ec5334fb5ab01576ce619c6b3e6094da25966fe1f2d5e04cR22-R27

statictype commented:
> The decoded values for Option<Some> and Option<None> look wrong to me, wdyt @statictype 

It's correct actually, the `Option` returns nothing in the `Some` branch

peetzweg commented:
> > The decoded values for Option and Option look wrong to me, wdyt @statictype
> 
> It's correct actually, the `Option` returns nothing in the `Some` branch

Ah true, I forgot to return something else than the unit value. I adapted the contract and updated this branch. Will return a bool for Ok Result and Some Option.

Furthermore I removed the code related to the Mapping type. Afaik an ink message is not able to return a whole mapping only individual values or an array.
